### PR TITLE
Source management sorting

### PR DESCRIPTION
**Pull request by Saelmala**
This PR:

- Adds a lastConnected to the Source dB schema, which updates on inventory syncing if the source is found in the api
- Adds functionality to the FilterDropdown to sort sources based on their lastConnected date, with a button to reverse the order of the sorting

**No sorting selected (in this case, the 'reverse sort order' button is disabled):**

**Sort on most recent is selected, reverse button NOT pressed:**

**Sort on most recent is selected, with the reverse button pressed:**

### CODE REVIEW

**Review by bwallberg — CHANGES_REQUESTED** [review]
I think for this to feel complete we should add the "lastConnected" date to the inventory card. Difficult to make an educated decision since the information isn't available 🙈

**Comment by Saelmala** [comment]
Added last connected date to the source card/ edit view:

File: src/api/manager/job/syncInventory.ts [hunk 1]
```diff
@@ -67,26 +73,30 @@ export async function runSyncInventory() {
       // If source was not found in response from API, always mark it as gone
       return { ...inventorySource, status: 'gone' } satisfies WithId<Source>;
     }
-    // Keep all old fields from the inventory source (name, tags, id, audio_stream etc), but update the status
+    // Keep all old fields from the inventory source (name, tags, id, audio_stream etc), but update the status and set the lastConnected to the current date
     return {
       ...inventorySource,
-      status: apiSource.status
+      status: apiSource.status,
+      lastConnected: new Date()
```
Comment (bwallberg) [inline src/api/manager/job/syncInventory.ts:80]: I just learned ( yesterday ) that the API now actually stores _some_ data about previously connected sources ( they used to just disappear...  ) so we have check the status here and only update lastConnected _if_ `status !== "gone"`.

**Review by bwallberg — CHANGES_REQUESTED** [review]
See comment about when to set lastConnected. The rest of code LGTM!

**Review by bwallberg — APPROVED** [review]
👏

**Review by malmen237 — APPROVED** [review]
LGTM

**Comment by LucasMaupin** [comment]
LGTM

**Pull request merged**